### ISSUE TITLE
How to understand the settings in vae?

### ISSUE THREAD

**Issue by Alex-zzh** [ZJUFanLab/scCube#4]
Hi,

In the definition of the vae class

```
class VAE(nn.Module):
        ...
        self.enc_feature_size_list = [self.embedding_size] + self.hidden_size_list + [self.mid_hidden * 2]
        self.dec_feature_size_list = [self.embedding_size] + self.hidden_size_list + [self.mid_hidden]
```

combine with 

```
  def reparameterize(self, mu, logvar):
      std = torch.exp(0.5 * logvar) # var => std
      eps = torch.randn_like(std)
      return eps * std + mu

  def forward(self, x, used_device):
      x = x.to(used_device)
      encoder_output = self.encode(x)
      mu, sigma = torch.chunk(encoder_output, 2, dim=1)  # mu, log_var
      hidden = self.reparameterize(mu, sigma)
      x_hat = self.decode(hidden)
      kl_div = 0.5 * torch.sum(torch.exp(sigma) + torch.pow(mu, 2) - 1 - sigma) / (x.shape[0] * x.shape[1])
      return x_hat, kl_div
```

means first half of the expression used as mu and second half used as sigma, I'd like to ask two questions:

1.Why do you choose this way?
2.Will it miss some expression info?I mean maybe it affected by the order of the genes when get hidden and x_hat.

Thank you for any response!

**Comment by ForwardYang98** [comment]
Hi,
This is a conventional code implementation of the VAE model.  **μ** and **σ** represent the mean and s.d. of of the approximate posterior, and through these two parameters, we can sample **ϵ** from the standard normal distribution and then compute the hidden **z** = **μ** + **σ** ⊙ **ϵ**. This reparameterization trick ensures that gradient can be backward propagated for updating the model parameters. Detailed explanations of this process can be found in the methods section of our paper.  Furthermore, the order of genes does not affect the quality of the generated data, as the generation process for each gene is independent.

**Comment by Alex-zzh** [comment]
Thanks for your feedback,I will go through it again.
